### PR TITLE
FE-1150 RSSI info between gateway & station in WiFi models

### PR DESCRIPTION
**Pull request by PavlosTze**
### **Why?**
We need to receive the D1 station's signal and show a warning or error message based on the signal's RSSI.

### **How?**
- Added the respective properties in the API Model.
- Added `LOW_STATION_RSSI` in `DeviceAlertType`
- Added a clickable `TextView` in the `MessageCardView` below the message as per the design.
- In the `UIDeviceInfoItem` replaced the previous `warning` and `ActionType` with `DeviceAlert` in order to properly handle every alert with its severity and its type.

### **Testing**
Check station Frontal Butter Windstorm or play with local mock and edit the file `get_user_device_info.json` and ensure everything looks OK. 
 <div id='description'>
<h3>Summary by Bito</h3>
This PR implements RSSI monitoring for WiFi weather stations, particularly the D1 model. It adds new fields to the WeatherStation model, introduces a LOW_STATION_RSSI alert type, and updates UI components. The code refactors to a more flexible DeviceAlert system and enhances the UI with troubleshooting options for low RSSI situations.
<br>
<br>
<b>Code change type</b>: Feature Addition, Refactoring, UI/UX Improvement
<br>
<br>
<b>Unit tests added</b>: False
<br>
<br>
<b>Estimated effort to review (1-5, lower is better)</b>: 3
</div>

### CODE REVIEW

File: app/src/main/java/com/weatherxm/ui/devicesettings/wifi/DeviceSettingsWifiViewModel.kt [hunk 1]
```diff
@@ -137,6 +140,25 @@ class DeviceSettingsWifiViewModel(
                 )
             }
 
+            val lastStationRssiTs =
+                stationRssiLastActivity?.getFormattedDateAndTime(context) ?: String.empty()
+            stationRssi?.let {
+                val deviceAlert = if (true) {
```
Comment (pantelisss) [inline app/src/main/java/com/weatherxm/ui/devicesettings/wifi/DeviceSettingsWifiViewModel.kt:146]: I think you missed something 😉

File: app/src/main/java/com/weatherxm/ui/devicesettings/wifi/DeviceSettingsWifiViewModel.kt [hunk 2]
```diff
@@ -137,6 +140,25 @@ class DeviceSettingsWifiViewModel(
...
+                    DeviceAlert.createError(DeviceAlertType.LOW_STATION_RSSI)
+                } else if (it in -95..-81) {
+                    DeviceAlert.createWarning(DeviceAlertType.LOW_STATION_RSSI)
+                } else {
+                    null
+                }
```
Comment (pantelisss) [inline app/src/main/java/com/weatherxm/ui/devicesettings/wifi/DeviceSettingsWifiViewModel.kt:152]: IMO, something like this looks cleaner
```
                val deviceAlert = if (it >= -80) {
                    null
                } else if (it >= -95) {
                    DeviceAlert.createWarning(DeviceAlertType.LOW_STATION_RSSI)
                } else {
                    DeviceAlert.createError(DeviceAlertType.LOW_STATION_RSSI)
                }
 
```

**Review by pantelisss — CHANGES_REQUESTED** [review]
Good job! Just check my comments

File: app/src/main/java/com/weatherxm/ui/devicesettings/wifi/DeviceSettingsWifiViewModel.kt (hunk 2 above)
Comment (PavlosTze) [reply app/src/main/java/com/weatherxm/ui/devicesettings/wifi/DeviceSettingsWifiViewModel.kt]: Agreed, good catches!

**Review by pantelisss — APPROVED** [review]
👏

**Comment by PavlosTze** [comment]
@CodiumAI-Agent /review

**Comment by PavlosTze** [comment]
AI Code Review is in progress (usually takes <3 minutes unless it's a very large PR).

**Comment by PavlosTze** [comment]
<div>
<h2>Changelist by <a href="https://bito.ai">Bito</a></h2>
<p>This pull request implements the following key changes.</p>

<div>
<table width="100%">
<tbody>
  <tr>
    <th>Key Change</th>
    <th>Files Impacted</th>
  </tr>

<tr>
<td width="30%">
    <b>Feature Improvement</b> - Enhanced Device Status Monitoring
</td>
<td width="70%">


<div>

-&nbsp;[`ApiModels.kt`](https://github.com/WeatherXM/wxm-android/pull/338/files#diff-c789267ebaada315b639750f7b9180af59cce98cbd072f8c221e6f2140ce450e) - Added stationRssi and stationRssiLastActivity fields to WeatherStation model
<div>

-&nbsp;[`UIModels.kt`](https://github.com/WeatherXM/wxm-android/pull/338/files#diff-75f457e8fab5ed68b9a0791417bc3d3fefc0c66ef7b34d882d60c61eb3d243ca) - Added LOW_STATION_RSSI to DeviceAlertType enum
<div>

-&nbsp;[`UIModels.kt`](https://github.com/WeatherXM/wxm-android/pull/338/files#diff-495f265f29351669b4249858f2861bd7cb9c7ea184cebd9beb09635cd29235d5) - Replaced action and warning fields with deviceAlert in UIDeviceInfoItem
<div>

-&nbsp;[`DeviceSettingsWifiViewModel.kt`](https://github.com/WeatherXM/wxm-android/pull/338/files#diff-45d344329d2d3499149f02587bb363367b16d63b6a92ce76687129a564eadded) - Implemented logic to determine device alert based on station RSSI
<div>

-&nbsp;[`DeviceSettingsHeliumViewModel.kt`](https://github.com/WeatherXM/wxm-android/pull/338/files#diff-ba72e2d6734de05f1740c5b7be1356d5646520e87d8f95e1e95980bc151ddc6f) - Updated device info generation to use DeviceAlert instead of UIDeviceAction
</div>
</td>
</tr>

<tr>
<td width="30%">
    <b>Other Improvements</b> - Enhanced User Interface for Device Alerts
</td>
<td width="70%">


<div>

-&nbsp;[`Views.kt`](https://github.com/WeatherXM/wxm-android/pull/338/files#diff-e82dd38683937a19b6e2a33b87e95a7f776106d6d484863ec29d398a2157f3cf) - Added clearMargins function for TextView
<div>

-&nbsp;[`MessageCardView.kt`](https://github.com/WeatherXM/wxm-android/pull/338/files#diff-68664248a1fd76408cf7ab42e501a151408c3e3f28a83b061122511d04ea84b8) - Added troubleshootMessage and clearMessageMargin functions
<div>

-&nbsp;[`DeviceInfoItemAdapter.kt`](https://github.com/WeatherXM/wxm-android/pull/338/files#diff-21dcd1d5390c506966cb365434347ea471ce9ccf3c5edc6337d2d7e43618bb6e) - Updated adapter to handle new DeviceAlert system and LOW_STATION_RSSI alert
<div>

-&nbsp;[`DeviceSettingsWifiActivity.kt`](https://github.com/WeatherXM/wxm-android/pull/338/files#diff-a58d3ffaffa88f3de20b986f53d8de1748390c0f457b2ecd0e50e65291fb76ba) - Implemented handling of LOW_STATION_RSSI alert with troubleshooting URL
</div>
</td>
</tr>

<tr>
<td width="30%">
    <b>Other Improvements</b> - Refactored Device Alert Handling
</td>
<td width="70%">


<div>

-&nbsp;[`DeviceAlertsAdapter.kt`](https://github.com/WeatherXM/wxm-android/pull/338/files#diff-61af4be8b8b7f5da3b1cc972f06675c5ca7d371f27feafcf43678a846a219d20) - Added handling for LOW_STATION_RSSI alert type
<div>

-&nbsp;[`BaseDeviceSettingsViewModel.kt`](https://github.com/WeatherXM/wxm-android/pull/338/files#diff-ef35bd40ae8ce804876eebd52fada3b67413053379752e7c09ea7c251b2fbf6e) - Updated to use DeviceAlert instead of warning string
<div>

-&nbsp;[`DeviceSettingsHeliumActivity.kt`](https://github.com/WeatherXM/wxm-android/pull/338/files#diff-14405dc24a3c51eec878b5994ccacc9afaac1a28337c7d985164dda517d130ec) - Updated to use DeviceAlertType instead of ActionType
<div>

-&nbsp;[`DevicesAdapter.kt`](https://github.com/WeatherXM/wxm-android/pull/338/files#diff-c715ca23617f444f1396d1702a7945d2bab52baa6cdc859484eaeff1fd831b0b) - Added handling for LOW_STATION_RSSI alert type
</div>
</td>
</tr>

</tbody>
</table>
</div>
</div>

**Pull request merged**